vue/src/components/UploadModelRun.vue

```diff
@@ -0,0 +1,200 @@
+<script setup lang="ts">
+import { reactive, ref } from "vue";
+import S3FileFieldClient, {
+  S3FileFieldResultState,
+} from "django-s3-file-field";
+import { ApiService } from "../client";
+
+const emits = defineEmits(['upload']);
+
+const s3ffClient = new S3FileFieldClient({
+  baseUrl: `${ApiService.getApiPrefix()}/s3-upload/`,
+  apiConfig: {
+    // django csrf token handling
+    xsrfCookieName: "csrftoken",
+    xsrfHeaderName: "X-CSRFToken",
+  },
+});
+
+const modelRun = reactive({
+  title: "",
+  region: "",
+  performer: "",
+  private: false,
+});
+
+const successDialog = ref(false);
+const uploadDialog = ref(false);
+const validForm = ref(false);
+const uploadFile = ref<File | File[]>();
+const uploadLoading = ref(false);
+const uploadError = ref<unknown>();
+// Infinity means indeterminate
+const uploadProgress = ref<number>(0);
+
+const QUERY_TASK_POLL_DELAY = 1000; // msec
+// from celery.states.READY_STATES
+const CELERY_READY_STATES = ["FAILURE", "SUCCESS", "REVOKED"];
+
+function openUploadDialog() {
+  uploadDialog.value = true;
+  uploadFile.value = undefined;
+  uploadError.value = undefined;
+}
+
+async function untilTaskReady(taskId: string) {
+  return new Promise((resolve) => {
+    async function queryTask() {
+      const result = await ApiService.getModelRunUploadTaskStatus(taskId);
+
+      if (CELERY_READY_STATES.includes(result)) {
+        resolve(result);
+      } else {
+        setTimeout(() => {
+          queryTask();
+        }, QUERY_TASK_POLL_DELAY);
+      }
+    }
+
+    queryTask();
+  });
+}
+
+async function upload() {
+  try {
+    uploadLoading.value = true;
```

Comment:
Can we use this to disable the inputs while the upload is processing?

There are some notes further down about the dialog being persistent and if we could allow multiple uploads processing at once in the celery tasks.  I think for now the additional code to do that is a bit much so just disabling the inputs when the upload is processing is enough.

Reply:
Done!